infinispan-embedded/src/main/java/org/hibernate/ogm/datastore/infinispan/util/ReflectionHelper.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.ogm.datastore.infinispan.util;
+
+import java.beans.BeanInfo;
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Reflection utilities.
+ *
+ * @author The Viet Nguyen &amp;ntviet18@gmail.com&amp;
+ */
+public class ReflectionHelper {
```

Comment:
There is a class with the same name in ogm core: `org.hibernate.ogm.util.impl.ReflectionHelper`

Could you add new methods to that one maybe? I have some additional comments I'll add below.